src/procedures/fulltext_create_index.c

```diff
@@ -17,13 +18,113 @@
 // CALL db.idx.fulltext.createNodeIndex(index_name, label, attributes)
 // CALL db.idx.fulltext.createNodeIndex('books', ['Book'], ['title', 'authors'])
 
+typedef struct {
+    Graph *g;
+    Schema *s;
+} _fulltextIndexCtx;
+
+static void _populateIndex (
+    RSIndex *idx,
+    Graph *g,
+    int label_id,
+    char **fields,
+    uint *fields_ids,
+    uint fields_count)
+{
+    Node node;
+    NodeID node_id;
+    GraphEntity *entity;
+    EntityProperty *prop;
+    GxB_MatrixTupleIter *it;
+    const GrB_Matrix label_matrix = Graph_GetLabel(g, label_id);
+    GxB_MatrixTupleIter_new(&it, label_matrix);
+
+    while(true) {
+        bool depleted = false;
+        GxB_MatrixTupleIter_next(it, NULL, &node_id, &depleted);
+        if(depleted) break;
+        
+        double score = 0;
+        const char* lang = NULL;
+        Graph_GetNode(g, node_id, &node);
+        RSDoc *doc = RediSearch_CreateDocument(&node_id, sizeof(EntityID), score, lang);
+
+        for(uint i = 0; i < fields_count; i++) {
+            RedisModuleString* s;
+            SIValue *v = GraphEntity_GetProperty((GraphEntity*)&node, fields_ids[i]);
+            if(v == PROPERTY_NOTFOUND) continue;
+            if(v->type & SI_NUMERIC) continue;
+            RediSearch_DocumentAddFieldString(doc, fields[i], v->stringval, strlen(v->stringval), RSFLDTYPE_FULLTEXT);
+        }
+
+        RediSearch_SpecAddDocument(idx, doc);
+    }
+
+    GxB_MatrixTupleIter_free(it);
+}
+
+static int _getNodeAttribute(void* ctx, const char* fieldName, const void* id, char** strVal, double* doubleVal) {
+    Node n;
+    NodeID nId = *(NodeID*)id;
+    _fulltextIndexCtx *indexCtx = ctx;
+    Graph *g = indexCtx->g;
+    Schema *s = indexCtx->s;
+    
+    Graph_GetNode(g, nId, &n);
+    Attribute_ID attrId = Schema_GetAttributeID(s, fieldName);
+    SIValue *v = GraphEntity_GetProperty((GraphEntity*)&n, attrId);
+    int ret;
+    if(v == PROPERTY_NOTFOUND) {
```

Comment:
Maybe skip booleans here, otherwise they should have handling.

Reply:
👍